server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java

```diff
@@ -465,6 +465,70 @@ void capabilityShouldNotAdvertiseLoginOnUnEncryptedChannel() throws Exception {
         }
     }
 
+    @Nested
+    class AuthenticationRequireSSL {
+        IMAPServer imapServer;
+
+        @AfterEach
+        void tearDown() {
+            if (imapServer!=null){
+                imapServer.destroy();
+            }
+        }
+
+        @Disabled("Waiting to https://github.com/apache/james-project/pull/791")
```

Comment:
Why do we need this? #791 already had tests for its own.
BTW I do not see tests for this PR. Maybe you can based on this work: https://github.com/apache/james-project/pull/770 ? 
```
-> if authentication.requireSSL is true and I issue a LOGIN command on an unencrypted channel it fails
-> if authentication.requireSSL is true and I issue a LOGIN command on an encrypted channel it succeed
-> if authentication.requireSSL is false and I issue a LOGIN command on an unencrypted channel it succeed
-> if authentication.requireSSL is false and I issue a LOGIN command on an encrypted channel it succeed
```
Also a test for fallback case?

Reply:
Ticket DOD define it https://github.com/linagora/james-project/issues/4441
`Add the same exact tests for AuthenticationRequireSSL`